test/unit/manipulation.js

```diff
@@ -1620,6 +1620,8 @@ function testHtml( valueObj ) {
 	tmp = " " + tmp.slice( 1 );
 	equal( div.html(valueObj( tmp )).html().replace( />/g, "&gt;" ), tmp, "Escaped html, leading space" );
 
+	equal( jQuery("<tr-d></tr-d><tr-d></tr-d>").length, 2, "Create tag-hyphenated elements." );
```

Comment:
And maybe a case with two (or more) hyphens, just in case. It's obvious current implementation will work in the same way but it's better to be future-proof.